### PR TITLE
Add dedicated product name to OCP\Defaults

### PR DESCRIPTION
**Pull request by juliusknorr**
There were some discrepancies between the public interface (where the name was considered [to be the instance name](https://github.com/nextcloud/server/blob/master/lib/public/Defaults.php#L109)) in contrast to the legacy OC_Defaults backend where it was used [as the software name](https://github.com/nextcloud/server/blob/cb057829f72c70e819f456edfadbb29d72dba832/lib/private/legacy/OC_Defaults.php#L179). In regular cases we should always be able to have a dedicated instance name (e.g. MyPersonalCloud) while keeping Nextcloud as a product name so this PR adds a separate method to obtain the product name as a separate label.

For branding scenarios the product name can be changed by using either a custom theme or a theming app config value, though not to publicly exposed as for most setups just saying Nextcloud there would be fine.

In addition to the new method I tried to adapt the usages that I've found referring to the software name.

### CODE REVIEW

File: lib/public/Defaults.php [hunk 1]
```diff
@@ -106,14 +106,24 @@ public function getDocBaseUrl() {
 	}
 
 	/**
-	 * name of your ownCloud instance
```
Comment (skjnldsv) [inline lib/public/Defaults.php:109]: :see_no_evil:

**Review by skjnldsv — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]
🐘

**Comment by rullzer** [comment]
CI says no

**Comment by blizzz** [comment]
CI is unhappy and assets have to be rebuild

**Comment by MorrisJobke** [comment]
Fix for a regression of this PR is in #27529

**Pull request merged**